app/src/main/java/com/fbada/projectone/domain/Car.kt

```diff
@@ -0,0 +1,572 @@
+                in 0..60 -> LOW
+                in 61..80 -> MEDIUM
+                in 81..100 -> HIGH
+                else -> ULTIMATE
+            }
+        }
+    }
+
+    /**
+     * Subclass Cougar Model type car.  Extends [Ford] class.
+     * @property radioType Type of radio inside car.
+     */
+    abstract class Cougar(
+        private val radioType: AvailableRadios,
+        seats: Int,
+        interior: AvailableInteriorMaterials,
+        owners: Int,
+        year: Int,
+        availableColours: AvailableColours,
+        fuelType: AvailableFuelTypes,
+        gearbox: AvailableGearBoxes
+    ) : Ford(seats, interior, owners, year, availableColours, fuelType, gearbox) {
+        /**
+         * @property price Base price plus interior score
+         */
+        override var price: Double = super.price + interiorPremium() * 15
+
+        /**
+         * Calculates a price based on interior materials and radio.
+         */
+        private fun interiorPremium(): Double {
+            val materialPriceScore = when (interiorMaterial) {
+                AvailableInteriorMaterials.LEATHER, AvailableInteriorMaterials.SUEDE -> HIGH
+                AvailableInteriorMaterials.COMBO -> HIGH
+                AvailableInteriorMaterials.FABRIC -> MEDIUM
+                else -> {
+                    LOW
+                }
+            }
+            val radioPriceScore = when (radioType) {
+                AvailableRadios.SMART_PHONE_CONNECT -> ULTIMATE
+                AvailableRadios.INTERNET, AvailableRadios.DIGITAL -> HIGH
+                AvailableRadios.CD_PLAYER -> LOW
+                else -> {
+                    MIN
+                }
+            }
+            return (materialPriceScore + radioPriceScore) * 0.75
+        }
+    }
+
+    /**
+     * Subclass Explorer Model type car.  Extends [Ford] class.
+     * @property spareTyre whether spare tyre available
+     * @property tyreSize Size of tyres.
+     */
+    abstract class Explorer(
+        var spareTyre: Boolean,
+        var tyreSize: Double,
+        seats: Int,
+        interior: AvailableInteriorMaterials,
+        owners: Int,
+        year: Int,
+        availableColours: AvailableColours,
+        fuelType: AvailableFuelTypes,
+        gearbox: AvailableGearBoxes
+    ) : Ford(seats, interior, owners, year, availableColours, fuelType, gearbox) {
+        /**
+         * @property price Explorer base price
+         */
+        override var price: Double = 700.00
+
+        /**
+         * Opens the boot
+         */
+        override fun openBoot() {
+            println("Rear access enabled")
+        }
+
+        /**
+         * Checks if a spare tyre is available based on its presence and size.
+         * @return true if a spare tyre is available and its size is less than 89, otherwise unavailable .
+         */
+        fun spareTyreAvailable(): Boolean {
+            return if (spareTyre) {
+                tyreSize < 89
+                true
```

Comment:
This condition will always return true even if the tyreSize will be 89 or more than 89. That will happen because after `tyreSize < 89` execution ends you explicitly return `true`
Personally I'd with something like:
```
fun spareTyreAvailable(): Boolean {
            return (tyreSize < 89).takeIf { spareTyre } ?: false
        }
```
Also based on the naming it's not clear that `tyreSize` is the `spareTyre` size.